packages/core/src/modules/credentials/formats/indy/IndyCredentialFormatService.ts

```diff
@@ -211,13 +212,19 @@ export class IndyCredentialFormatService extends CredentialFormatService<IndyCre
   public async processOffer({ attachment, credentialRecord }: FormatProcessOptions) {
     this.logger.debug(`Processing indy credential offer for credential record ${credentialRecord.id}`)
 
-    const credOffer = attachment.getDataAsJson<Indy.CredOffer>()
+    const credentialOffer = attachment.getDataAsJson<Indy.CredOffer>()
 
-    if (!credOffer.schema_id || !credOffer.cred_def_id) {
+    if (!credentialOffer.schema_id || !credentialOffer.cred_def_id) {
       throw new CredentialProblemReportError('Invalid credential offer', {
         problemCode: CredentialProblemReportReason.IssuanceAbandoned,
       })
     }
+
+    credentialRecord.metadata.set(CredentialMetadataKeys.IndyCredential, {
+      credentialDefinitionId: credentialOffer.cred_def_id,
+      schemaId: credentialOffer.schema_id,
+    })
+    await this.credentialRepository.update(credentialRecord)
```

Comment:
Updating is done in the credential service after tje credential format has been called. The formats services just modify it. Were you running into issues with things not being saved?